components/script/dom/htmlimageelement.rs

```diff
@@ -1032,3 +1047,197 @@ fn image_dimension_setter(element: &Element, attr: LocalName, value: u32) {
     let value = AttrValue::Dimension(value.to_string(), dim);
     element.set_attribute(&attr, value);
 }
+
+#[derive(Debug)]
+#[derive(PartialEq)]
+pub struct ImageSource {
+    pub url: String,
+    pub descriptor: Descriptor,
+}
+#[derive(PartialEq)]
+#[derive(Debug)]
+pub struct Descriptor {
+    pub wid: Option<u32>,
+    pub den: Option<f64>,
+}
+
+pub fn collect_sequence_characters<F>(s: &str, predicate: F) -> (&str, &str)
+        where F: Fn(&char) -> bool
+        {
+            for (i, ch) in s.chars().enumerate() {
+                if !predicate(&ch) {
+                    return (&s[0..i], &s[i..])
+                }
+            }
+        return (s, "");
+        }
+
+// https://html.spec.whatwg.org/multipage/#parsing-a-srcset-attribute.
+pub fn parse_a_srcset_attribute(input: String) -> Vec<ImageSource> {
+    let mut start = 0;
+    let mut candidates: Vec<ImageSource> = vec![];
+    while start < input.len() {
+        let position = &input[start..];
+        let(spaces, position) = collect_sequence_characters(position, |c| *c ==',' || char::is_whitespace(*c));
+        let x = spaces.find(',');
+        match x {
+            Some(val) => println!("Parse Error"),
+            None => println!("No commas\n"),
```

Comment:
We should remove the println! calls now. Since a parse error does not actually change the way this code behaves, we can remove the check for the commas here.